client.go

```diff
@@ -108,6 +108,10 @@ func (c *Client) GetEnvironmentFlags(ctx context.Context) (f Flags, err error) {
 	return Flags{}, &FlagsmithClientError{msg: fmt.Sprintf("Failed to fetch flags with error: %s", err)}
 }
 
+type GetIdentityFlagsOpts struct {
+	Transient bool `json:"transient,omitempty"`
+}
```

Comment:
@gagantrivedi Keen to know if this approach is redundant. Maybe we could just go with a `transient bool` argument to respective interfaces.

Reply:
Yeah, I'd prefer the latter approach

Reply:
Having implemented the `bool` argument, I don't quite like not being able to provide `nil` as the value. Since Golang doesn't support optional args, I feel we're forcing the majority of users to make the decision they don't need to make.

Reply:
I think a bool pointer should be able to achieve that?

Reply:
It should, in theory, but then there's no way to provide a pointer to a constant (that I know of, at least).

Reply:
e.g. this will work: 
```go
client.GetIdentityFlags(ctx, "test_identity", nil, nil)
``` 
but this will fail:
```go
client.GetIdentityFlags(ctx, "test_identity", nil, &false)
```

Reply:
ugh, yeah true. I guess if we have to major version this anyway, I'd prefer to implement it like open feature: https://github.com/open-feature/go-sdk?tab=readme-ov-file#usage

Reply:
We can also implement a different method for transient identity as well :thinking:

Reply:
> We can also implement a different method for transient identity as well 🤔

This might be a good solution, yes

Reply:
> So a custom struct? I believe that's what I did initially?

Yes, but that custom struct will also take identifier and traits

Reply:
> Yes, but that custom struct will also take identifier and traits

I like that approach a lot. I'll mark the PR as breaking and implement this, then.

Reply:
> > Yes, but that custom struct will also take identifier and traits
> 
> I like that approach a lot. I'll mark the PR as breaking and implement this, then.

I'd like to hear what @matthewelwell thinks re this because we will have to do that across all the SDK(typed, at least)

Reply:
My personal opinion is no, we don't have to this to other SDKs, at least for languages that support default argument values. But sure, let's wait for team input before we move forward.

Reply:
Sorry, I'm just trying to fully understand the issue here. As I understand it, the issue is about adding the argument to determine if an identity is transient or not? The issue being that we can't just add a new argument that defaults to `false` because golang doesn't support this, so we'd have to force people to set it to `false` explicitly? Unless I'm missing something (which I quite possibly am) that sounds like it would contribute a breaking change regardless? 

The original solution was a custom struct of just the `opts` which would include the `transient` property. 

The solution suggested by @gagantrivedi is to create a new struct that includes all of the arguments to the method, including the identifier and the traits. 

Is that correct? 

I'm interested to know what the benefit is of creating this new struct to include all of the arguments. Is it just for cleanliness of the interface?

Reply:
@matthewelwell Users can provide `nil` in lieu of the struct, so they don't have to make a decision they don't need to make. Moreover, each struct member can be `nil`, so it can be used to provide "optional" / non-essential arguments. This was my initial train of thought.

Gagan offered to use a struct for the rest of the arguments in order to not complicate the interface signature. However, he'd prefer to do it across all typed SDKs.

Reply:
> However, he'd prefer to do it across all typed SDKs.
I don't think this is the correct pattern to implement in other SDks, e.g. Java so I certainly don't think we should consider doing this across all typed SDKs. 

In terms of the solution for the Golang SDK, I'm leaning towards the original implementation of the `opts` struct, and maybe we add an issue tracking changes that we want to make in the next major version?

Reply:
> he'd prefer to do it across all typed SDKs.

We can choose not to do that, but then we will have to live with different interfaces across different SDKs.

Reply:
> In terms of the solution for the Golang SDK, I'm leaning towards the original implementation of the `opts` struct, and maybe we add an issue tracking changes that we want to make in the next major version?

The original implementation is a breaking change. We will either have to major version this or add another function/method

Reply:
> I don't think this is the correct pattern to implement in other SDks, e.g. Java 

Why though? Just so we are on the same page, I am talking about implementing the interface that OpenFeature has.

Reply:
What you're suggesting, were we to implement in Java, would look something like the following, right? 

```java
public Flags getIdentityFlags(GetIdentityFlagsInputs inputs) {
    ...
}
```

That just (to me) doesn't feel like idiomatic java.

Reply:
> What you're suggesting, were we to implement in Java, would look something like the following, right?
> 
> ```java
> public Flags getIdentityFlags(GetIdentityFlagsInputs inputs) {
>     ...
> }
> ```
> 
> That just (to me) doesn't feel like idiomatic java.

There won't be `getIdentityFlags` it would just be `getFlags`

```
public Flags getFlags(EvaluationContext ctx) {
    ...
}
```

Reply:
Like this: https://github.com/open-feature/java-sdk?tab=readme-ov-file#targeting

Reply:
Hmm... yep, ok, that's a good point. Perhaps, for now, we should add that method and mark `getEnvironmentFlags` and `getIdentityFlags` as deprecated? To use transient identities, you will need to move to use the `getFlags` method.

Reply:
I also like implementing the OF interface with an evaluation context that can contain the identity. Actually, an immediate benefit of doing that is that we could optionally send the environment or another environment key in this call providing an easy way to "switch context" I guess, but that's just an example and another discussion.

Reply:
Annoyingly I don't think it'll be as straight forward on the client-side with OF. In JS there's a [single OF context](https://github.com/open-feature/js-sdk/blob/main/packages/client/README.md#targeting-and-context). I wonder if this could be achieved using their concept of [domains](https://github.com/open-feature/js-sdk/blob/main/packages/client/README.md#domains).

Reply:
@kyle-ssg Domain-specific contexts look like the exact solution for this.